tests/test_qaneeded.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from bugbug import bugzilla
+from bugbug.models.qaneeded import QANeededModel
+
+
+def test_get_qaneeded_labels():
+    model = QANeededModel()
+    classes = model.get_labels()
+    assert not classes[1389220]
+    assert classes[1389223]
+    assert classes[1390433]
+
+
+def test_rollback():
+    model = QANeededModel()
+
+    bugs = {}
+    for bug in bugzilla.get_bugs():
+        bugs[int(bug['id'])] = bug['history']
+
+    def test(bug_id):
```

Comment:
The name of the function is too generic.

BTW, we also want to test that we are rolling back to the right point in the history

Reply:
To test if rollback is to the right time, should I check if all comments are made before the `rollback_date`?

Reply:
You could simply assert on the number of changes you loop through before finding the right one.

Reply:
Basically make count a integer increasing it for every iteration of the inner loop, then in the caller do:
`assert rollback_point(BUG_ID) == X`

Reply:
Sorry for the delay, just finished with my exams. Correct me if I'm wrong, but rollback is supposed to take a bug and return the same bug in an initial stage, right? Because whenever I test the function, the value before and after rollback always comes as equal.

Reply:
Rollback is used to go back in the history of a bug, not exclusively to the initial state, but also to a given state (this is what the `rollback_when` function is for, to know at which point in the history to stop).

In your case, if a bug has qawanted in the history, rollback should go back to the point when qawanted was added.